phac-nml/irida-next
# Analysis Exports: Multiple Selection UI

## What does this PR do and why?
This PR adds the UI to select multiple workflow executions for export.
Main changes include:
- Checkboxes and select/deselect all feature for workflow execution tables
- Fix data export preview tab for analysis exports to include multiple workflow executions
- Refactor SampleActions (for listing) to ListActions to be inclusive of listing both samples and workflow executions

## Screenshots or screen recordings
[Screencast from 2024-08-27 02:39:22 PM.webm](https://github.com/user-attachments/assets/ef2302cc-8442-483f-b314-ed1b22355d84)

## How to set up and validate locally
1. Set up the ability to run workflow executions locally as described in [PR 507](https://github.com/phac-nml/irida-next/pull/507)
2. Run a couple workflow executions
3. Navigate to the user workflow executions page and test the select/deselect all buttons and ensure they behave as expected.
4. Select multiple workflow executions and create an export
5. In the dialog, verify that the selection number and listed workflow execution data is accurate.
6. Submit the export
7. After 30s, download the export and verify the export contents are as expected.
8. Click the `Preview` tab and ensure the page listing is as expected.
9. Repeat the above with workflow executions from `Automated Workflow Executions`, The workflows can be set up with the steps described in [PR585](https://github.com/phac-nml/irida-next/pull/585). 
10. Create an export from a workflow execution's `show` page, and verify steps 5-8.

Other tests:
- Verify that the other data export, transfer samples, and clone samples dialogs still list the samples appropriately as the logic for this was refactored

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
